oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/worker/MetricsPersistentWorker.java

```diff
@@ -174,21 +175,25 @@ private void flushDataToStorage(List<Metrics> metricsList,
                     /*
                      * Merge metrics into cachedMetrics, change only happens inside cachedMetrics.
                      */
-                    cachedMetrics.combine(metrics);
-                    cachedMetrics.calculate();
-                    prepareRequests.add(metricsDAO.prepareBatchUpdate(model, cachedMetrics));
-                    nextWorker(cachedMetrics);
+                    isAbandoned = !cachedMetrics.combine(metrics);
+                    if (isAbandoned) {
+                        cachedMetrics.calculate();
+                        prepareRequests.add(metricsDAO.prepareBatchUpdate(model, cachedMetrics));
+                        nextWorker(cachedMetrics);
```

Comment:
Should be add `else continue` here, rather than adding `if (!isAbandoned) ` at L190?